shared/teams/team/settings/retention/warning/container.js

```diff
@@ -8,17 +8,22 @@ const mapStateToProps = (state: TypedState, {routeProps}) => {
   }
 }
 
-const mapDispatchToProps = (dispatch: Dispatch, {routeProps, navigateUp}) => ({
-  onBack: () => dispatch(navigateUp()),
-  // TODO maybe take some more identifiers {teamname, conversationIDKey} and handle it here
-  // when you're not decreasing the policy this warning doesn't show so the parent component
-  // needs to know how to make the calls anyway
-  onConfirm: () => {
-    const cb: () => void = routeProps.get('onConfirm')
-    cb && cb()
-    dispatch(navigateUp())
-  },
-})
+const mapDispatchToProps = (dispatch: Dispatch, {routeProps, navigateUp}) => {
+  const onBack = () => dispatch(navigateUp())
+  return {
+    onBack,
+    onCancel: () => {
+      onBack()
+      const onCancel: () => void = routeProps.get('onCancel')
```

Comment:
typed as being non null, likely need ? if you're afraid onCancel can be undefined